File: src/main/java/org/jukeboxmc/potion/FatalPoison.java
package org.jukeboxmc.potion;

import org.jukeboxmc.entity.EntityLiving;
import org.jukeboxmc.event.entity.EntityDamageEvent;

import java.awt.*;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class FatalPoison extends Effect {

    @Override
    public int getId() {
        return 25;
    }

    @Override
    public EffectType getEffectType() {
        return EffectType.FATAL_POISON;
    }

    @Override
    public Color getEffectColor() {
        return new Color( 78, 147, 49 );
    }

    @Override
    public void apply( EntityLiving entityLiving ) {
        if ( this.canExecute() ) {
            if ( entityLiving.getHealth() > 2 ) {
                entityLiving.damage( new EntityDamageEvent( entityLiving, 1, EntityDamageEvent.DamageSource.MAGIC_EFFECT ) );
            }
        }
    }

    @Override
    public boolean canExecute() {
        int interval;
        if ( ( interval = ( 25 >> this.amplifier ) ) > 0 ) {
            return ( this.duration % interval ) == 0;
        }
        return false;
    }

    @Override
    public void update( long currentTick ) {

    }

    @Override
    public void remove( EntityLiving entityLiving ) {

    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/effect/FatalPoison.kt
package org.jukeboxmc.server.effect

import org.jukeboxmc.api.effect.EffectType
import org.jukeboxmc.api.event.entity.EntityDamageEvent
import org.jukeboxmc.server.entity.JukeboxEntityLiving
import java.awt.Color

class FatalPoison : JukeboxEffect() {

    override fun getId(): Int {
        return 25
    }

    override fun getEffectType(): EffectType {
        return EffectType.FATAL_POISON
    }

    override fun getEffectColor(): Color {
        return Color(78, 147, 49)
    }

    override fun apply(entityLiving: JukeboxEntityLiving) {
        if (this.canExecute()) {
            if (entityLiving.getHealth() > 2) {
                entityLiving.damage(EntityDamageEvent(entityLiving, 1F, EntityDamageEvent.DamageSource.MAGIC_EFFECT))
            }
        }
    }

    override fun update(currentTick: Long) {

    }

    override fun remove(entityLiving: JukeboxEntityLiving) {

    }

    override fun canExecute(): Boolean {
        val interval: Int
        if ((25 shr this.getAmplifier()).also { interval = it } > 0) {
            return (this.getDuration() % interval) == 0
        }
        return false
    }
}